src/TabWidget.js

```diff
@@ -0,0 +1,386 @@
+        domConstruct.place(this.moreTabList, parentNode);
+      }
+    }
+    return this;
+  },
+  /**
+   * Handler for positioning the focus bar for the tab list.
+   * @param {Object} The target tab in the tabList.
+   */
+  positionFocusState: function positionFocusState(target) {
+    const focusState = query('.animated-bar', this.id)[0];
+
+    if (focusState) {
+      const posTop = target.offsetTop;
+      const posLeft = target.offsetLeft;
+      const width = parseInt(target.offsetWidth, 10);
+      const height = parseInt(target.offsetHeight, 10);
+      const tableTop = this.tabList.offsetTop;
+      const tableLeft = this.tabList.offsetLeft;
+
+      domStyle.set(focusState, {
+        left: posLeft - tableLeft + 'px',
+        top: posTop - tableTop + 'px',
+        right: (posTop - tableTop) + width + 'px',
+        bottom: (posTop - tableTop) + height + 'px',
+        width: width + 'px',
+      });
+    }
+    return this;
+  },
+  /**
+   * Checks the tab to see if it causes an overflow when placed in the tabList, if so then push it a new list element called More.
+   * @param {Object} The tab object.
+   */
+  checkTabOverflow: function checkTabOverflow(tab) {
+    if (tab.offsetTop > this.tabList.offsetTop) {
+      if (!this.inOverflow) {
+        const moreTab = domConstruct.toDom(this.moreTabItemTemplate.apply({
+          title: this.moreText + '...',
+        }, this));
+        domStyle.set(moreTab, { // eslint-disable-line
+          float: 'right',
+        });
+        domConstruct.place(moreTab, this.tabList);
+
+        this.tabMoreIndex = array.indexOf(this.tabList.children, tab);
+        this.tabList.children[this.tabMoreIndex].remove();
+        if (this.tabList.children.length === 1 && !this.moreTabList.children.length) {
+          domClass.add(moreTab, 'selected');
+          this.currentTab = tab;
+          domClass.toggle(tab, 'selected');
+        }
+
+        if (moreTab.offsetTop > this.tabList.offsetTop) {
+          this.tabMoreIndex = this.tabMoreIndex - 1;
+          const replacedTab = this.tabList.children[this.tabMoreIndex];
+          this.tabList.children[this.tabMoreIndex].remove();
+          domConstruct.place(replacedTab, this.moreTabList);
+        }
+
+        domConstruct.place(tab, this.moreTabList);
+        this.inOverflow = true;
+        this.tabMoreIndex++;
+      } else {
+        this.tabList.children[this.tabMoreIndex].remove();
+        domConstruct.place(tab, this.moreTabList);
+      }
+    }
+    return this;
+  },
+  /**
+   * Function used to create the tabs, should be called by the parent upon completion of populating the tabs array of dom objects
+   * @param {Array} An array of the tab objects.
+  */
+  createTabs: function createTabs(tabs = []) {
+    array.forEach(tabs, function placeTab(tab) {
+      if (this.defaultTabIndex >= 0) {
+        if (this.defaultTabIndex === this.tabList.children.length) {
+          this.currentTab = tab;
```

Comment:
We are setting the current tab in multiple location with the same duplicated code.